src/libecalc/common/utils/rates.py

```diff
@@ -152,11 +152,28 @@ class Config:
         extra = Extra.forbid
 
     @validator("values", each_item=True, pre=True)
-    def convert_none_to_nan(cls, v: float, field: ModelField) -> TimeSeriesValue:
+    def convert_none_to_nan(cls, v: float, values, field: ModelField) -> TimeSeriesValue:
         if field.outer_type_ is float and v is None:
             return math.nan
         return v
 
+    @root_validator(pre=True)
+    def timesteps_values_one_to_one(cls, field_values):
+        nr_timesteps = len(field_values["timesteps"])
+        nr_values = len(field_values["values"])
+
+        if not cls.__name__ == TimeSeriesVolumes.__name__:
+            if nr_timesteps != nr_values:
+                if all(math.isnan(i) for i in field_values["values"]):
+                    # TODO: This should be solved another place. Temporary solution to make things run
+                    field_values["values"] == [math.nan] * len(field_values["timesteps"])
+                else:
+                    raise ProgrammingError(
+                        "Time series: number of timesteps do not match number " "of values. Most likely a bug."
```

Comment:
Report to eCalc Dev Team :)

Reply:
Done (had to create a new PR, rebasing stuff...)

Reply:
New PR: https://github.com/equinor/ecalc/pull/251